Sources/SwiftlyCore/HTTPClient.swift

```diff
@@ -68,14 +162,39 @@ public struct SwiftlyHTTPClient {
 
     /// Return an array of released Swift versions that match the given filter, up to the provided
     /// limit (default unlimited).
-    ///
-    /// TODO: retrieve these directly from swift.org instead of through GitHub.
     public func getReleaseToolchains(
+        platform: PlatformDefinition,
+        arch a: String? = nil,
         limit: Int? = nil,
         filter: ((ToolchainVersion.StableRelease) -> Bool)? = nil
     ) async throws -> [ToolchainVersion.StableRelease] {
-        let filterMap = { (gh: GitHubTag) -> ToolchainVersion.StableRelease? in
-            guard let release = try gh.parseStableRelease() else {
+        let arch = if a == nil {
+#if arch(x86_64)
+            "x86_64"
+#elseif arch(arm64)
+            "aarch64"
+#else
+            #error("Unsupported processor architecture")
+#endif
+        } else {
+            a!
+        }
+
+        let url = "https://swift.org/api/v1/install/releases.json"
+        let swiftOrgReleases: [SwiftOrgRelease] = try await self.getFromJSON(url: url, type: [SwiftOrgRelease].self)
+
+        var swiftOrgFiltered: [ToolchainVersion.StableRelease] = swiftOrgReleases.compactMap { swiftOrgRelease in
+            guard let swiftOrgPlatform = swiftOrgRelease.platforms.first(where: { $0.platformName == platform.name || platform.name == "xcode" }) else {
+                return nil
+            }
+
+            guard swiftOrgPlatform.archs.contains(arch) || platform.name == "xcode" else {
+                return nil
+            }
```

Comment:
I guess this code is just assuming there'll be a release for "xcode".

Reply:
In each SwiftOrgRelease there is a list of platforms with their archs. The exception is xcode, which doesn't have an entry in that platform list. I'll see if I can rewrite this to be clearer.